### PR TITLE
[MBL-17282][Student] Redesign calendar event details screen

### PR DESCRIPTION
**Pull request by kristofnemere**
Test plan: 
- same functionality as before
- slightly different design (check Figma)
- Edit/Delete will handled in another ticket
- Test opening this screen from all the places: ToDo list, Important Dates, Syllabus, Schedule, Calendar

refs: MBL-17282
affects: Student
release note: none

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/EventViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        return DateHelper.getFormattedDate(context, startDate).orEmpty()
+    }
+
+    fun handleAction(action: EventAction) {
+        when (action) {
+            is EventAction.OnLtiClicked -> viewModelScope.launch {
+                _events.send(EventViewModelAction.OpenLtiScreen(action.url))
+            }
+
+            is EventAction.DeleteEvent -> {}
```
Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/EventViewModel.kt:139]: Pretty much same here, I would try and complete this screen with the events as much as possible just to make it easier to integrate everything else.

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                        .fillMaxSize(),
+                )
+            },
+            modifier = modifier
+        )
+    }
+}
+
+@Composable
+private fun TopAppBarContent(
```
Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt:99]: It's not just the content only, bit misleading.

**Review by hermannakos — APPROVED** [review]
QA + 1

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/EventViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        canvasContext?.backgroundColor?.let { color ->
+            _uiState.update { it.copy(toolbarColor = color) }
+        }
+    }
+
+    private suspend fun updateUiStateWithData(scheduleItem: ScheduleItem) {
+        _uiState.update {
+            it.copy(
+                calendar = scheduleItem.contextName.orEmpty(),
+                modifyAllowed = scheduleItem.contextId == apiPrefs.user?.id,
```
[the remark this answers is not in the document]
Comment (kristofnemere) [reply libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/EventViewModel.kt:101]: I'll find this out when I can test it, probably in the next ticket - currently it's not working properly even on web

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt [hunk 4]
```diff
@@ -96,44 +95,45 @@ internal fun EventScreen(
...
+            if (uiState.toolbarUiState.modifyAllowed) {
                 OverFlowMenuSegment(
                     actionHandler = actionHandler,
                     showDeleteConfirmationDialog = showDeleteConfirmationDialog
                 )
             }
         },
-        backgroundColor = Color(uiState.toolbarColor),
+        backgroundColor = Color(uiState.toolbarUiState.toolbarColor),
         contentColor = Color.White,
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/calendarevent/details/composables/EventScreen.kt:134]: I would use `colorResource(R.color.white)` to be consistent.

**Review by tamaskozmer — APPROVED** [review]

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Pull request merged**